### DISCUSSION TITLE
Default for pluralization

### DISCUSSION THREAD

**Discussion by nkoehring** [ivanhofer/typesafe-i18n#545] [DX improvements]
A locale with pluralization is transformed into a function of the shape `(arg0: string | number | boolean) => LocalizedString`. It would be very convenient to make arg0 optional if it is not used inside the template, and return a default when undefined.

This would allow the following:

```ts
LLL('House{{s}}') // 'House'
LLL('House{{s}}', 23) // 'Houses'
LLL('{n} House{{s}}', 23) // '23 Houses'
LLL('{n} House{{s}}') // Error, because n is used in template string.
```

The obvious question coming to mind now is: What would be the default? I think, `one` is a good candidate.

**Comment by ivanhofer** [comment]
Thanks for your suggestion!
With that approach you would probably miss some parts in your code where you should call the function with pluralization. Making it a required prop ensures you think about the plural case you want to use.

Think of this localized string in your code : 'Houses'.
Some times later you find a bug and you realize that you need it in its singular variant for some parts of your project.
You decide to change it to `House{{s}}`.
Then you would need to check your full code base if you call the translation function the correct way.
And there you probably will miss to update everything accordingly.
This will not happen if the param is required.

`typesafe-i18n` once also had the feature to write `LL.hello` instead of `LL.hello()`  for translations without params. There the decision to get rid of the simpler syntax was similar: if you change the translation, you probably won't catch all errors because `TypeScript` can't tell you about them.
